### ISSUE TITLE
Custom Loss function for Blender?

### ISSUE THREAD

**Issue by josharnoldjosh** [facebookresearch/ParlAI#2917]
Hello,

I was wondering how I could implement a custom loss function for fine-tuning the Blender model?

I already have the loss function, I am just not sure how to implement it exactly. 

Would I override the `BertPreTrainedModel` class in order to do this? Or for example the `BertForMaskedLM` class? My loss will be subclassed from `torch.nn` just like `MSELoss` is.

Is there any high-level guidance that could be provided as to how to do this? E.g where I should look in the code for an example or idea? 

Thanks!

**Comment by stephenroller** [comment]
Inherit from TransformerGeneratorAgenr and override compute_loss.

It needs to be document.

**Comment by josharnoldjosh** [comment]
Thank you!
